pkgs/applications/networking/mailreaders/claws-mail/default.nix

```
@@ -1,97 +1,171 @@
   };
 
+  # The corresponding commit in the gtk3 branch. There is no git tag.
+  gtk3src = {
+    rev = "a5741530b9ab16224c64c2a5a6af79b0bf671454";
+    sha256 = "13705rw40hl3j3lkfrlzjvj00yj7lng6ljqd4vprwzkq3qm5gnbs";
+  };
+in stdenv.mkDerivation rec {
+  pname = "claws-mail";
+  inherit version;
+
+  src = fetchgit ({
+    url = "git://git.claws-mail.org/claws.git";
+  } // (if useGtk3 then gtk3src else gtk2src));
+
   outputs = [ "out" "dev" ];
 
-  patches = [
-    ./mime.patch
-  ];
+  patches = [ ./mime.patch ];
 
   preConfigure = ''
     # autotools check tries to dlopen libpython as a requirement for the python plugin
     export LD_LIBRARY_PATH=$LD_LIBRARY_PATH''${LD_LIBRARY_PATH:+:}${python}/lib
+    # generate version without .git
+    [ -e version ] || echo "echo ${version}" > version
   '';
 
   postPatch = ''
     substituteInPlace src/procmime.c \
         --subst-var-by MIMEROOTDIR ${shared-mime-info}/share
   '';
 
-  nativeBuildInputs = [ autoreconfHook pkgconfig wrapGAppsHook python.pkgs.wrapPython ];
+  nativeBuildInputs = [ autoreconfHook pkgconfig bison flex wrapGAppsHook python.pkgs.wrapPython ];
   propagatedBuildInputs = with python.pkgs; [ python ] ++ optionals enablePluginPython [ pygtk pygobject2 ];
 
   buildInputs =
-    [ curl dbus dbus-glib gtk2 gnutls gsettings-desktop-schemas
-      libetpan perl glib-networking libSM libytnef
-    ]
-    ++ optional enableSpellcheck enchant
-    ++ optionals (enablePgp || enablePluginSmime) [ gnupg gpgme ]
-    ++ optional enablePluginArchive libarchive
-    ++ optional enablePluginNotificationSounds libcanberra-gtk2
-    ++ optional enablePluginNotificationDialogs libnotify
-    ++ optional enablePluginLitehtmlViewer gumbo
-    ++ optional enablePluginRssyl libxml2
-    ++ optional enableNetworkManager networkmanager
-    ++ optional enableLdap openldap
-    ++ optional enablePluginPdf poppler
-    ++ optional enablePluginVcalendar libical;
+    [ curl gsettings-desktop-schemas glib-networking ]
+    ++ [(if useGtk3 then gtk3 else gtk2)]
+    ++ optional  enableLibSM libSM
+    ++ optional  enableGnuTLS gnutls
+    ++ optional  enableEnchant enchant
+    ++ optionals enableDbus [ dbus dbus-glib ]
+    ++ optional  enableLdap openldap
+    ++ optional  enableNetworkManager networkmanager
+    ++ optional  enableLibetpan libetpan
+    ++ optional  enableValgrind valgrind
+    ++ optional  enableSvg librsvg
+    ++ optional  enablePluginArchive libarchive
+    ++ optional  enablePluginGdata libgdata
+    ++ optional  enablePluginLitehtmlViewer gumbo
+    ++ optionals enablePluginNotification [ libnotify ] ++ [(if useGtk3 then libcanberra-gtk3 else libcanberra-gtk2)]
+    ++ optional  enablePluginPerl perl
+    ++ optional  enablePluginPdfViewer poppler
+    ++ optional  enablePluginRssyl libxml2
+    ++ optionals enablePluginPgp [ gnupg gpgme ]
+    ++ optional  enablePluginTnefParse libytnef
+    ++ optional  enablePluginVcalendar libical;
 
```

Nitpick, that might make future diffs cleaner by a bit:

```suggestion
    ++ optional  enablePluginVcalendar libical
  ;
```